src/lib/hooks/swap/useSendSwapTransaction.tsx

```diff
@@ -29,6 +29,8 @@ interface FailedCall extends SwapCallEstimate {
   error: Error
 }
 
+class SwapError extends Error {}
```

Comment:
perhaps name this `InvalidSwapError` or something. such that it's clear it's prior to executing the swap?